### PR TITLE
[FLINK-18457][docs] Fix invalid links in "Detecting Patterns" page of "Streaming Concepts"

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

*Fix invalid links in "Detecting Patterns" page of "Streaming Concepts"*

## Brief change log

Fix invalid links in "Detecting Patterns" page of "Streaming Concepts" 
 The page location is : `flink/docs/dev/table/streaming/match_recognize.md`.
 The page url is: [Patterns in Tables](https://ci.apache.org/projects/flink/flink-docs-master/dev/table/streaming/match_recognize.html#time-attributes).
 The Invalid link is [interval joins](https://ci.apache.org/projects/flink/flink-docs-master/dev/table/streaming/match_recognize.html#joins), however, the cerrect link target should be linked as  [interval joins](https://ci.apache.org/projects/flink/flink-docs-master/dev/table/streaming/joins.html#interval-joins)
The corresponding Chinese page will be repaired , which is  reported in [FLINK-16087](https://issues.apache.org/jira/browse/FLINK-16087 ).

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi, @alpinegizmo .

I have fixed the invalid links in `flink/docs/dev/table/streaming/match_recognize.md`. 

And I will fixed the invalid links in `flink/docs/dev/table/streaming/match_recognize.zh.md` in PR(https://github.com/apache/flink/pull/12798).

Could you help me to review this PR(https://github.com/apache/flink/pull/12802) if you have free time?

Thank you for your help.

Best,
Roc Marshal.

**Comment by RocMarshal** [comment]
Hi, @wuchong  .

I have fixed the invalid links in `flink/docs/dev/table/streaming/match_recognize.md`. 

And I will fix the invalid links in `flink/docs/dev/table/streaming/match_recognize.zh.md` in PR(https://github.com/apache/flink/pull/12798).

Could you help me to review this PR(https://github.com/apache/flink/pull/12802) if you have free time?

Thank you .

Best,
Roc Marshal.

File: docs/dev/table/streaming/match_recognize.md [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ use [time attributes](time_attributes.html). To select those there are available
       <td><p>Returns the timestamp of the last row that was mapped to the given pattern.</p>
       <p>The resulting attribute is a <a href="time_attributes.html">rowtime attribute</a>
          that can be used in subsequent time-based operations such as
-         <a href="#joins">interval joins</a> and <a href="#aggregations">group window or over
+         <a href="joins.html#interval-joins">interval joins</a> and <a href="#aggregations">group window or over
```
Comment (wuchong) [inline docs/dev/table/streaming/match_recognize.md:1041]: Please use `% link %` tag.

Comment (RocMarshal) [reply docs/dev/table/streaming/match_recognize.md]: Done. @wuchong

File: docs/dev/table/streaming/match_recognize.md [hunk 2]
```diff
@@ -1038,7 +1038,7 @@ use [time attributes](time_attributes.html). To select those there are available
       <td><p>Returns the timestamp of the last row that was mapped to the given pattern.</p>
       <p>The resulting attribute is a <a href="time_attributes.html">rowtime attribute</a>
          that can be used in subsequent time-based operations such as
-         <a href="joins.html#interval-joins">interval joins</a> and <a href="#aggregations">group window or over
+         <a href="{% link joins.md#interval-joins %}">interval joins</a> and <a href="#aggregations">group window or over
```
Comment (wuchong) [inline docs/dev/table/streaming/match_recognize.md:1041]: I guess this is an invalid link. Could you try to `build_docs.sh` in your local to verify this?

Comment (RocMarshal) [reply docs/dev/table/streaming/match_recognize.md]: Yes, @wuchong . You are right.
I'm so sorry for this mistake !
I corrected the invalid links based on your suggestions and tips of `build_docs.sh`.
Thank you for your reminding !

**Pull request merged**